test/functional/tests/io_class/test_io_class_stats_file_size_core.py

```diff
@@ -0,0 +1,218 @@
+                        check_statistics(cache, core, io_classes, io_class, issued_reqs_no)
+                        fs_utils.remove(f"{core.mount_point}/*", force=True, recursive=True)
+                    size_min = size + Size(512, Unit.Byte)
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_io_class_stats_file_size_core_direct(cache_mode):
+    """
+    title: Open CAS statistics values per core for IO classification by file size - direct IO.
+    description: Check Open CAS ability to assign correct IO class when IO is issued in direct mode.
+    pass_criteria:
+        - after FIO with direct statistics increases only for direct IO class
+    """
+
+    with TestRun.step("Prepare devices."):
+        cache_device, core_devices = prepare_devices(cores_count=1)
+
+    with TestRun.step("Start cache and add core devices (one for each filesystem)."):
+        cache, core = prepare_cache_and_cores(cache_device, cache_mode, core_devices)
+
+    with TestRun.step("In default IO class configuration file set Eviction priority=1 "
+                      "and load it for all caches."):
+        io_classes = prepare_io_classes(cache)
+        *file_size_based_io_classes, io_class_direct = io_classes
+
+    with TestRun.step("Run fio for all devices checking IO class statistics."):
+        sizes = prepare_fio_sizes(file_size_based_io_classes)
+        size_min = Size(512, Unit.Byte)
+
+        for io_class, size in zip(file_size_based_io_classes, sizes):
+            with TestRun.step(f"Testing IO class {io_class.id}."):
+                with TestRun.step("Run fio for each device with direct=1 and check that only "
+                                  "for direct IO class statistics increases."):
+                    core.reset_counters()
+                    fio = fio_params(core, size_min, size, direct=True)
+                    result = fio.run(fio_timeout=timedelta(minutes=5))
+                    sync()
+                    drop_caches()
+                    issued_reqs_no = \
+                        result[0].write_requests_number() + result[0].read_requests_number()
+                    check_statistics(cache, core, io_classes, io_class_direct, issued_reqs_no)
+                    fs_utils.remove(f"{core.path}/*", force=True, recursive=True)
+                    size_min = size + Size(512, Unit.Byte)
+
+
+def fio_params(core, size_min, size_max, direct=False):
+    name_size_min = core.path if direct else f"{core.mount_point}/{round(size_min.get_value())}"
+    name_size_max = core.path if direct else f"{core.mount_point}/{round(size_max.get_value())}"
+    fio = Fio().create_command() \
+        .io_engine(IoEngine.libaio) \
+        .time_based() \
+        .run_time(timedelta(seconds=90) if size_max <= Size(1, Unit.GibiByte)
+                  else timedelta(seconds=200)) \
```

Comment:
maybe instead of time_based runs, jobs would have io_size = x * file_size (x even as low as 1.1), which would write data > than file size, but not extend the execution time very much (especially for low-sized files)